diff --git a/Makefile b/Makefile
index bcdb148..98e1079 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ GITTAG = v$(VERSION)
 
 DIRS = doc doc/examples contrib tuningplugins monitorplugins ktune udev libexec
 FILES = tuned tuned.spec Makefile tuned.py tuned.initscript tuned.conf tuned-adm tuned_adm.py tuned-adm.pam tuned-adm.consolehelper tuned_nettool.py tuned_logging.py tuned.bash tuned.tmpfiles
-FILES_doc = doc/DESIGN.txt doc/README.utils doc/TIPS.txt doc/tuned.8 doc/tuned.conf.5 doc/tuned-adm.1 doc/README.scomes doc/diskdevstat.8 doc/netdevstat.8 doc/scomes.8 doc/varnetload.8
+FILES_doc = doc/DESIGN.txt doc/README.utils doc/TIPS.txt doc/tuned.8 doc/tuned.conf.5 doc/tuned-adm.1 doc/README.scomes doc/diskdevstat.8 doc/netdevstat.8 doc/scomes.8 doc/varnetload.8 doc/tuned-profiles.7 doc/tuned-profiles-sap.7 doc/tuned-profiles-sap-hana.7
 FILES_examples = ktune/sysctl.ktune
 FILES_contrib = contrib/diskdevstat contrib/netdevstat contrib/scomes contrib/varnetload
 FILES_tuningplugins = tuningplugins/cpu.py tuningplugins/disk.py tuningplugins/net.py tuningplugins/__init__.py
@@ -115,6 +115,10 @@ install:
 	install -m 0644 doc/tuned.conf.5 $(DESTDIR)/usr/share/man/man5
 	mkdir -p $(DESTDIR)/usr/share/man/man1
 	install -m 0644 doc/tuned-adm.1 $(DESTDIR)/usr/share/man/man1
+	mkdir -p $(DESTDIR)/usr/share/man/man7
+	install -m 0644 doc/tuned-profiles.7 $(DESTDIR)/usr/share/man/man7
+	install -m 0644 doc/tuned-profiles-sap.7 $(DESTDIR)/usr/share/man/man7
+	install -m 0644 doc/tuned-profiles-sap-hana.7 $(DESTDIR)/usr/share/man/man7
 
 	# Install ktune
 	install -m 755 -d $(DESTDIR)/etc
diff --git a/doc/tuned-adm.1 b/doc/tuned-adm.1
index 246f12b..53a6b7f 100644
--- a/doc/tuned-adm.1
+++ b/doc/tuned-adm.1
@@ -1,7 +1,7 @@
 .\"/* 
 .\" * All rights reserved
-.\" * Copyright (C) 2009 Red Hat, Inc.
-.\" * Authors: Marcela Mašláňová, Phil Knirsch
+.\" * Copyright (C) 2009-2014 Red Hat, Inc.
+.\" * Authors: Marcela Mašláňová, Phil Knirsch, Jaroslav Škarvada
 .\" *
 .\" * This program is free software; you can redistribute it and/or
 .\" * modify it under the terms of the GNU General Public License
@@ -18,7 +18,7 @@
 .\" * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 .\" */
 .\" 
-.TH TUNED_ADM "1" "9 Jul 2009" "Linux Programmer's Manual"
+.TH TUNED_ADM "1" "15 Oct 2014" "Linux Programmer's Manual"
 .SH NAME
 tuned-adm \- commandline tool for switching between different tuning profiles
 .SH SYNOPSIS
@@ -29,8 +29,6 @@ tuned-adm \- commandline tool for switching between different tuning profiles
 .SH DESCRIPTION
 This command line utility allows user to switch between user definable tuning profiles. Several predefined profiles are already included for some of the more common cases. Although the profiles typically refer to a common use case you can of course switch to any profile on any system. So switching to the laptop-battery-powersave profile on a server will activate this profile on it. This way, depending on your requirements you can select any of the profiles that fits your needs. You can even create your own profile, either based on one of the existing ones by copying it or make a completely new one. The profiles are stored in subdirectories below /etc/tune-profiles.
 
-The power saving profiles contain settings that are typically not enabled by default as they will noticeably impact the latency/performance of your system as opposed to the power saving mechanisms we have enabled by default. On the other hand the performance profiles disable the additional power saving mechanisms of tuned and ktune as they would negatively impact the performance tuning, be it throughput or latency.
-
 .SH "OPTIONS"
 
 .SS
@@ -54,60 +52,17 @@ Switches to the given profilename. If none is given or no valid one is given the
 .B off
 Switch off all ktune and tuned settings and remove them from services. The configuration file are set to previous settings.
 
-.SH PREDEFINED PROFILES
-At the moment we're providing the following pre defined profiles:
-
-.TP
-.BI "default"
-This is the default powersaving profile. It is the lowest of the available profiles in regard to powersaving and only enables CPU and disk plugins of tuned. This is not the same as 'tuned-adm off', where tuned and ktune services are disabled.
-
-.TP
-.BI "desktop-powersave"
-A powersaving profile directed at desktop systems. Enables ALPM powersaving for SATA host adapters as well as the CPU, ethernet and disk plugins of tuned.
-
-.TP
-.BI server-powersave
-A powersaving profile directed at server systems. Enables ALPM powersaving for SATA host adapters, disables CD-ROM polling via hal and activates the CPU and disk plugins of tuned.
-
-.TP
-.BI laptop-ac-powersave
-Medium powersaving profile directed at laptops running on AC. Enables ALPM powersaving for SATA host adapters,  WiFi power saving as well as CPU, ethernet and disk plugins of tuned.
-
-.TP
-.BI laptop-battery-powersave
-Strong powersaving profile directed at laptops running on battery. It activates all power saving mechanisms from the previous profiles plus it enables multi core power savings scheduler for low wakeup systems and makes sure the ondemand governor is active and the AC97 audio power saving is enabled.  This profile can be used for any kind of system, not only laptops on battery power in order to save the maximum amount of power. The tradeoff for this profile is a noticeable impact on performance, specifically latency of disk and network IO. 
-
-.TP
-.BI "spindown-disk"
-Strong powersaving profile enabling aggresive disk spin-down. Disk writeback values are increased and disk swappines is lowered. CD drives polling and log syncing is disabled. All partitions are remounted with 'noatime' option. All tuned plugins are disabled.
-
-.TP
-.BI throughput-performance
-Server profile for typical throughput performance tuning. Disables tuned and ktune power saving mechanisms and enables sysctl settings that improve the throughput performance of your disk and network IO and switches to the deadline scheduler. CPU governor is set to performance.
-
-.TP
-.BI latency-performance
-Server profile for typical latency performance tuning. Disables tuned and ktune power saving mechanisms and enables deadline IO scheduler. CPU governor is set to performance.
-
-.TP
-.BI "enterprise-storage"
-Server profile for high disk throughput performace tuning. Disables tuned and ktune power saving mechanisms and enables deadline IO scheduler. Enables hugepages and disables disk barriers. Disk readahead values are increased. CPU governor is set to performance.
-
-.TP
-.BI "virtual-guest"
-Profile optimized for virtual guests based on enterprise-storage profile. It additionally decreases virtual memory swapiness.
-
-.TP
-.BI "virtual-host"
-Profile optimized for virtual hosts based on enterprise-storage profile. It additionally decreases virtual memory swapiness and enables more aggresive writeback of dirty pages.
-
 .SH "FILES"
 .nf
 /etc/tune-profiles/*
 
 .SH "SEE ALSO"
 .BR tuned (8)
+.BR tuned-profiles (7)
+.BR tuned-profiles-sap (7)
+.BR tuned-profiles-sap-hana (7)
 .SH AUTHOR
-.nf
+.NF
 Marcela Mašláňová <mmaslano@redhat.com>
 Phil Knirsch <pknirsch@redhat.com>
+Jaroslav Škarvada <jskarvad@redhat.com>
diff --git a/doc/tuned-profiles-sap-hana.7 b/doc/tuned-profiles-sap-hana.7
new file mode 100644
index 0000000..2ead85a
--- /dev/null
+++ b/doc/tuned-profiles-sap-hana.7
@@ -0,0 +1,57 @@
+.\"/* 
+.\" * All rights reserved
+.\" * Copyright (C) 2009-2014 Red Hat, Inc.
+.\" * Authors: Jaroslav Škarvada
+.\" *
+.\" * This program is free software; you can redistribute it and/or
+.\" * modify it under the terms of the GNU General Public License
+.\" * as published by the Free Software Foundation; either version 2
+.\" * of the License, or (at your option) any later version.
+.\" *
+.\" * This program is distributed in the hope that it will be useful,
+.\" * but WITHOUT ANY WARRANTY; without even the implied warranty of
+.\" * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+.\" * GNU General Public License for more details.
+.\" *
+.\" * You should have received a copy of the GNU General Public License
+.\" * along with this program; if not, write to the Free Software
+.\" * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+.\" */
+.\" 
+.TH TUNED_PROFILES_SAP_HANA "7" "23 Sep 2014" "Fedora Power Management SIG" "tuned"
+.SH NAME
+tuned\-profiles\-sap\-hana - description of profiles provided for the SAP HANA
+
+.SH DESCRIPTION
+These profiles provides performance optimizations for the SAP HANA applications.
+
+.SH PROFILES
+The following profiles are provided:
+
+.TP
+.BI "sap\-hana"
+A performance optimized profile for the SAP HANA applications.
+It is based on throughput\-performance profile. It additionally disables
+transparent hugepages, locks CPU to the low C states (by PM QoS) and tunes sysctl
+regarding semaphores.
+
+.TP
+.BI "sap\-hana\-vmware"
+A performance optimized profile for the SAP HANA applications on VMware.
+It is based on throughput\-performance profile. It additionally disables
+transparent hugepages, locks CPU to the low C states (by PM QoS) and tunes sysctl
+regarding semaphores. It decreases virtual memory swappiness and disables large
+recieve offload for the network adapter.
+
+.SH "FILES"
+.NF
+/etc/tune\-profiles/*
+
+.SH "SEE ALSO"
+.BR tuned (8)
+.BR tuned\-adm (1)
+.BR tuned\-profiles (7)
+.BR tuned\-profiles\-sap (7)
+.SH AUTHOR
+.NF
+Jaroslav Škarvada <jskarvad@redhat.com>
diff --git a/doc/tuned-profiles-sap.7 b/doc/tuned-profiles-sap.7
new file mode 100644
index 0000000..4211775
--- /dev/null
+++ b/doc/tuned-profiles-sap.7
@@ -0,0 +1,49 @@
+.\"/* 
+.\" * All rights reserved
+.\" * Copyright (C) 2009-2014 Red Hat, Inc.
+.\" * Authors: Jaroslav Škarvada
+.\" *
+.\" * This program is free software; you can redistribute it and/or
+.\" * modify it under the terms of the GNU General Public License
+.\" * as published by the Free Software Foundation; either version 2
+.\" * of the License, or (at your option) any later version.
+.\" *
+.\" * This program is distributed in the hope that it will be useful,
+.\" * but WITHOUT ANY WARRANTY; without even the implied warranty of
+.\" * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+.\" * GNU General Public License for more details.
+.\" *
+.\" * You should have received a copy of the GNU General Public License
+.\" * along with this program; if not, write to the Free Software
+.\" * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+.\" */
+.\" 
+.TH TUNED_PROFILES_SAP "7" "23 Sep 2014" "Fedora Power Management SIG" "tuned"
+.SH NAME
+tuned\-profiles\-sap - description of profiles provided for the SAP NetWeaver
+
+.SH DESCRIPTION
+These profiles provides performance optimizations for the SAP NetWeaver applications.
+
+.SH PROFILES
+The following profiles are provided:
+
+.TP
+.BI "sap\-netweaver"
+A performance optimized profile for the SAP NetWeaver applications.
+It is based on throughput\-performance profile. It additionally tunes sysctl
+settings regarding shared memory, semaphores and maximum number of memory map
+areas a process may have.
+
+.SH "FILES"
+.NF
+/etc/tune\-profiles/*
+
+.SH "SEE ALSO"
+.BR tuned (8)
+.BR tuned\-adm (1)
+.BR tuned\-profiles (7)
+.BR tuned\-profiles\-sap\-hana (7)
+.SH AUTHOR
+.NF
+Jaroslav Škarvada <jskarvad@redhat.com>
diff --git a/doc/tuned-profiles.7 b/doc/tuned-profiles.7
new file mode 100644
index 0000000..fce7048
--- /dev/null
+++ b/doc/tuned-profiles.7
@@ -0,0 +1,94 @@
+.\"/* 
+.\" * All rights reserved
+.\" * Copyright (C) 2009-2014 Red Hat, Inc.
+.\" * Authors: Marcela Mašláňová, Phil Knirsch, Jaroslav Škarvada
+.\" *
+.\" * This program is free software; you can redistribute it and/or
+.\" * modify it under the terms of the GNU General Public License
+.\" * as published by the Free Software Foundation; either version 2
+.\" * of the License, or (at your option) any later version.
+.\" *
+.\" * This program is distributed in the hope that it will be useful,
+.\" * but WITHOUT ANY WARRANTY; without even the implied warranty of
+.\" * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+.\" * GNU General Public License for more details.
+.\" *
+.\" * You should have received a copy of the GNU General Public License
+.\" * along with this program; if not, write to the Free Software
+.\" * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+.\" */
+.\" 
+.TH TUNED_PROFILES "7" "23 Sep 2014" "Fedora Power Management SIG" "tuned"
+.SH NAME
+tuned\-profiles - description of basic tuned profiles
+
+.SH DESCRIPTION
+These are the base profiles which are mostly shipped in the base tuned
+package. They are targeted to various goals. Mostly they provide
+performance optimizations but there are also profiles targeted to
+low power consumption, low latency and others. You can mostly deduce the
+purpose of the profile by its name or you can see full description bellow.
+
+The power saving profiles contain settings that are typically not enabled by default as they will noticeably impact the latency/performance of your system as opposed to the power saving mechanisms we have enabled by default. On the other hand the performance profiles disable the additional power saving mechanisms of tuned and ktune as they would negatively impact the performance tuning, be it throughput or latency.
+
+.SH PREDEFINED PROFILES
+At the moment we're providing the following pre defined profiles:
+
+.TP
+.BI "default"
+This is the default powersaving profile. It is the lowest of the available profiles in regard to powersaving and only enables CPU and disk plugins of tuned. This is not the same as 'tuned-adm off', where tuned and ktune services are disabled.
+
+.TP
+.BI "desktop\-powersave"
+A powersaving profile directed at desktop systems. Enables ALPM powersaving for SATA host adapters as well as the CPU, ethernet and disk plugins of tuned.
+
+.TP
+.BI "server\-powersave"
+A powersaving profile directed at server systems. Enables ALPM powersaving for SATA host adapters, disables CD-ROM polling via hal and activates the CPU and disk plugins of tuned.
+
+.TP
+.BI "laptop\-ac\-powersave"
+Medium powersaving profile directed at laptops running on AC. Enables ALPM powersaving for SATA host adapters,  WiFi power saving as well as CPU, ethernet and disk plugins of tuned.
+
+.TP
+.BI "laptop\-battery\-powersave"
+Strong powersaving profile directed at laptops running on battery. It activates all power saving mechanisms from the previous profiles plus it enables multi core power savings scheduler for low wakeup systems and makes sure the ondemand governor is active and the AC97 audio power saving is enabled.  This profile can be used for any kind of system, not only laptops on battery power in order to save the maximum amount of power. The tradeoff for this profile is a noticeable impact on performance, specifically latency of disk and network IO. 
+
+.TP
+.BI "spindown\-disk"
+Strong powersaving profile enabling aggresive disk spin-down. Disk writeback values are increased and disk swappines is lowered. CD drives polling and log syncing is disabled. All partitions are remounted with 'noatime' option. All tuned plugins are disabled.
+
+.TP
+.BI "throughput\-performance"
+Server profile for typical throughput performance tuning. Disables tuned and ktune power saving mechanisms and enables sysctl settings that improve the throughput performance of your disk and network IO and switches to the deadline scheduler. CPU governor is set to performance.
+
+.TP
+.BI "latency\-performance"
+Server profile for typical latency performance tuning. Disables tuned and ktune power saving mechanisms and enables deadline IO scheduler. CPU governor is set to performance.
+
+.TP
+.BI "enterprise\-storage"
+Server profile for high disk throughput performace tuning. Disables tuned and ktune power saving mechanisms and enables deadline IO scheduler. Enables hugepages and disables disk barriers. Disk readahead values are increased. CPU governor is set to performance.
+
+.TP
+.BI "virtual\-guest"
+Profile optimized for virtual guests based on enterprise-storage profile. It additionally decreases virtual memory swapiness.
+
+.TP
+.BI "virtual\-host"
+Profile optimized for virtual hosts based on enterprise-storage profile. It additionally decreases virtual memory swapiness and enables more aggresive writeback of dirty pages.
+
+.SH "FILES"
+.NF
+/etc/tune\-profiles/*
+
+.SH "SEE ALSO"
+.BR tuned (8)
+.BR tuned\-adm (1)
+.BR tuned\-profiles\-sap (7)
+.BR tuned\-profiles\-sap\-hana (7)
+.SH AUTHOR
+.NF
+Marcela Mašláňová <mmaslano@redhat.com>
+Phil Knirsch <pknirsch@redhat.com>
+Jaroslav Škarvada <jskarvad@redhat.com>
diff --git a/ktune/README.ktune b/ktune/README.ktune
index 49358f4..16021c7 100644
--- a/ktune/README.ktune
+++ b/ktune/README.ktune
@@ -48,6 +48,7 @@ service is enabled and started, it will use the configuration options from
 
     SYSCTL="/etc/sysctl.ktune"
     USE_KTUNE_D="yes"
+    SYSCTL_PRE="/etc/sysctl.d/* /etc/sysctl.conf"
     SYSCTL_POST="/etc/sysctl.d/* /etc/sysctl.conf"
     ELEVATOR="deadline"
     ELEVATOR_TUNE_DEVS="/sys/block/{sd,cciss}*/queue/scheduler"
@@ -60,7 +61,12 @@ profiles from the /etc/ktune.d directory. A profile has to have the extension
 .conf to be used. If there is an additional file with the same name, but the
 extension .sh it will be used as a profile script.
 
-SYSCTL_POST references the system wide sysctl file for customer settings.
+SYSCTL_PRE references the system wide sysctl file for customer settings that
+is loaded before the ktune tunings. The tunings from this file may be
+overridden by ktune settings. Most profile don't use this option.
+
+SYSCTL_POST references the system wide sysctl file for customer settings that
+is loaded after the ktune tunings. This may override the ktune settings.
 
 ELEVATOR is the I/O scheduler, ktune will set for ELEVATOR_TUNE_DEVS. If you
 want to disable the setting of the I/O scheduler, just comment out the ELEVATOR
diff --git a/ktune/ktune.init b/ktune/ktune.init
index 76df2da..19ce0f1 100644
--- a/ktune/ktune.init
+++ b/ktune/ktune.init
@@ -55,9 +55,15 @@ $(LANG=C /bin/ls -1 /etc/ktune.d/*.conf 2>/dev/null)
 EOF
 fi
 
+function is_int
+{
+  [ ! -z "${1##*[^0-9]*}" ]
+}
+
 load_file() {
     local file=$1
     local save=$2
+    local curr_value
 
     ret_s=0
     errors=
@@ -69,19 +75,34 @@ load_file() {
 	key=$(echo $line | awk '{ match($0,"[ \t]*=[ \t]*") && $0=substr($0,1,RSTART-1); print $0}')
 	value=$(echo $line | awk '/=/ { match($0,"[ \t]*=[ \t]*"); $0=substr($0,RSTART+RLENGTH); print $0}')
 
+	curr_value=""
+	if [ "${value:0:1}" = '>' ]; then
+	    value=${value:1}
+	    curr_value=$($SYSCTL_GET -n "$key")
+	fi
+
+	if [ "${value:0:1}" = '$' ]; then
+	    value=${value:1}
+	    value=${!value}
+	fi
+
+	is_int "$curr_value" && is_int "$value" && [ "$value" -lt "$curr_value" ] && value="$curr_value"
+
 	[ $save -ne 0 ] && save_value=$($SYSCTL_GET "$key")
 
 	setting=$key
-	[ -n "$value" ] && setting+="=${value}"
+	if [ "$value" ]; then
+	    [ -n "$value" ] && setting+="=${value}"
 
-	error="$($SYSCTL_SET "${setting}" 2>&1)"
-	ret=$?
-	if [ $save -ne 0 -a $ret -eq 0 ] && ! grep -q "^${key} =" $SAVE_FILE 2>/dev/null; then
-	    echo "$save_value" >> "$SAVE_FILE"
-	fi
-	[ -n "$error" ] && errors+="${error}\n"
+	    error="$($SYSCTL_SET "${setting}" 2>&1)"
+	    ret=$?
+	    if [ $save -ne 0 -a $ret -eq 0 ] && ! grep -q "^${key} =" $SAVE_FILE 2>/dev/null; then
+	        echo "$save_value" >> "$SAVE_FILE"
+	    fi
+	    [ -n "$error" ] && errors+="${error}\n"
 
-	let ret_s+=$ret
+	    let ret_s+=$ret
+	fi
     done < "$file"
 
     if [ $ret_s -eq 0 ]; then
@@ -114,6 +135,14 @@ load_sysctl() {
     /bin/rm -f "$SAVE_FILE"
 
     if [ ${#KTUNE_FILES[@]} -gt 0 ]; then
+        # Apply general sysctl settings beforehand
+	for f in $SYSCTL_PRE; do
+	    if [ -r "$f" ]; then
+		echo $"Applying sysctl settings from $f"
+		sysctl -e -p "$f" >/dev/null 2>&1
+	    fi
+	done
+
 	echo $"Applying ktune sysctl settings:"
 
 	# Apply ktune files.
diff --git a/tune-profiles/sap-hana-vmware/ktune.sh b/tune-profiles/sap-hana-vmware/ktune.sh
new file mode 100755
index 0000000..1a987a9
--- /dev/null
+++ b/tune-profiles/sap-hana-vmware/ktune.sh
@@ -0,0 +1,23 @@
+#!/bin/sh
+
+. /etc/tune-profiles/functions
+
+start() {
+	set_cpu_governor performance
+	/usr/libexec/tuned/pmqos-static.py cpu_dma_latency=1
+	multiply_disk_readahead 4
+	ethtool -K eth0 lro off
+	set_transparent_hugepages never
+	return 0
+}
+
+stop() {
+	restore_cpu_governor
+	/usr/libexec/tuned/pmqos-static.py disable
+	restore_transparent_hugepages
+	restore_disk_readahead
+	ethtool -K eth0 lro on
+	return 0
+}
+
+process $@
diff --git a/tune-profiles/sap-hana-vmware/ktune.sysconfig b/tune-profiles/sap-hana-vmware/ktune.sysconfig
new file mode 100644
index 0000000..f0f08e7
--- /dev/null
+++ b/tune-profiles/sap-hana-vmware/ktune.sysconfig
@@ -0,0 +1,27 @@
+# ktune service configuration
+
+# This is the ktune sysctl file.  You can comment this out to prevent ktune
+# from applying its sysctl settings.
+#SYSCTL="/etc/sysctl.ktune"
+
+# Use *.conf files in the ktune configuration directory /etc/ktune.d.
+#   Value: yes|no,  default: yes
+# It is useful if you want to load settings from additional files. Set this to
+# no if you to prevent ktune from using these additional files.
+USE_KTUNE_D="yes"
+
+# This is the custom sysctl configuration file.  Any settings in this file will
+# be applied after the ktune settings, overriding them.  Comment this out to
+# use only the ktune settings.
+SYSCTL_POST="/etc/sysctl.d/* /etc/sysctl.conf"
+
+# This is the I/O scheduler ktune will use.  This will *not* override anything
+# explicitly set on the kernel command line, nor will it change the scheduler
+# for any block device that is using a non-default scheduler when ktune starts.
+# You should probably leave this on "deadline", but "as", "cfq", and "noop" are
+# also legal values.  Comment this out to prevent ktune from changing I/O
+# scheduler settings.
+ELEVATOR="noop"
+
+# These are the devices, that should be tuned with the ELEVATOR
+ELEVATOR_TUNE_DEVS="/sys/block/{sd,cciss,dm-,vd}*/queue/scheduler"
diff --git a/tune-profiles/sap-hana-vmware/sysctl.ktune b/tune-profiles/sap-hana-vmware/sysctl.ktune
new file mode 100644
index 0000000..8d09200
--- /dev/null
+++ b/tune-profiles/sap-hana-vmware/sysctl.ktune
@@ -0,0 +1,44 @@
+# ktune sysctl settings for rhel6 servers with enterprise-class
+# storage, maximizing i/o throughput
+#
+# Minimal preemption granularity for CPU-bound tasks:
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+kernel.sched_min_granularity_ns = 10000000
+
+# SCHED_OTHER wake-up granularity.
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+#
+# This option delays the preemption effects of decoupled workloads
+# and reduces their over-scheduling. Synchronous workloads will still
+# have immediate wakeup/sleep latencies.
+kernel.sched_wakeup_granularity_ns = 15000000
+
+# Filesystem I/O is usually much more efficient than swapping, so try to keep
+# swapping low.  It's usually safe to go even lower than this on systems with
+# server-grade storage.
+vm.swappiness = 30
+
+# If a workload mostly uses anonymous memory and it hits this limit, the entire
+# working set is buffered for I/O, and any more write buffering would require
+# swapping, so it's time to throttle writes until I/O can catch up.  Workloads
+# that mostly use file mappings may be able to use even higher values.
+#
+# The generator of dirty data starts writeback at this percentage (system default
+# is 20%)
+vm.dirty_ratio = 40
+
+# Start background writeback (via writeback threads) at this percentage (system
+# default is 10%)
+#vm.dirty_background_ratio = 15
+
+# PID allocation wrap value.  When the kernel's next PID value
+# reaches this value, it wraps back to a minimum PID value.
+# PIDs of value pid_max or larger are not allocated.
+#
+# A suggested value for pid_max is 1024 * <# of cpu cores/threads in system>
+# e.g., a box with 32 cpus, the default of 32768 is reasonable, for 64 cpus,
+# 65536, for 4096 cpus, 4194304 (which is the upper limit possible).
+#kernel.pid_max = 65536
+
+# This is for HANA deployment
+kernel.sem = 1250 256000 100 8192
diff --git a/tune-profiles/sap-hana-vmware/tuned.conf b/tune-profiles/sap-hana-vmware/tuned.conf
new file mode 100644
index 0000000..6c574fc
--- /dev/null
+++ b/tune-profiles/sap-hana-vmware/tuned.conf
@@ -0,0 +1,21 @@
+# disable tuned
+
+[main]
+
+[DiskMonitor]
+enabled=False
+
+[DiskTuning]
+enabled=False
+
+[NetMonitor]
+enabled=False
+
+[NetTuning]
+enabled=False
+
+[CPUMonitor]
+enabled=False
+
+[CPUTuning]
+enabled=False
diff --git a/tune-profiles/sap-hana/ktune.sh b/tune-profiles/sap-hana/ktune.sh
new file mode 100755
index 0000000..59a11c4
--- /dev/null
+++ b/tune-profiles/sap-hana/ktune.sh
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+# RHEL for SAP HANA ktune script
+# based on the enterprise-storage profile
+
+. /etc/tune-profiles/functions
+
+start() {
+	set_cpu_governor performance
+	/usr/libexec/tuned/pmqos-static.py cpu_dma_latency=1
+	set_transparent_hugepages never
+	multiply_disk_readahead 4
+
+	return 0
+}
+
+stop() {
+	restore_cpu_governor
+	/usr/libexec/tuned/pmqos-static.py disable
+	restore_transparent_hugepages
+	restore_disk_readahead
+
+	return 0
+}
+
+process $@
diff --git a/tune-profiles/sap-hana/ktune.sysconfig b/tune-profiles/sap-hana/ktune.sysconfig
new file mode 100644
index 0000000..7b38eb7
--- /dev/null
+++ b/tune-profiles/sap-hana/ktune.sysconfig
@@ -0,0 +1,30 @@
+# RHEL for SAP HANA ktune sysctl settings
+# based on the enterprise-storage profile
+
+# ktune service configuration
+
+# This is the ktune sysctl file.  You can comment this out to prevent ktune
+# from applying its sysctl settings.
+#SYSCTL="/etc/sysctl.ktune"
+
+# Use *.conf files in the ktune configuration directory /etc/ktune.d.
+#   Value: yes|no,  default: yes
+# It is useful if you want to load settings from additional files. Set this to
+# no if you to prevent ktune from using these additional files.
+USE_KTUNE_D="yes"
+
+# This is the custom sysctl configuration file.  Any settings in this file will
+# be applied after the ktune settings, overriding them.  Comment this out to
+# use only the ktune settings.
+SYSCTL_POST="/etc/sysctl.d/* /etc/sysctl.conf"
+
+# This is the I/O scheduler ktune will use.  This will *not* override anything
+# explicitly set on the kernel command line, nor will it change the scheduler
+# for any block device that is using a non-default scheduler when ktune starts.
+# You should probably leave this on "deadline", but "as", "cfq", and "noop" are
+# also legal values.  Comment this out to prevent ktune from changing I/O
+# scheduler settings.
+ELEVATOR="deadline"
+
+# These are the devices, that should be tuned with the ELEVATOR
+ELEVATOR_TUNE_DEVS="/sys/block/{sd,cciss,dm-,vd}*/queue/scheduler"
diff --git a/tune-profiles/sap-hana/sysctl.ktune b/tune-profiles/sap-hana/sysctl.ktune
new file mode 100644
index 0000000..1999f90
--- /dev/null
+++ b/tune-profiles/sap-hana/sysctl.ktune
@@ -0,0 +1,47 @@
+# RHEL for SAP HANA ktune sysctl settings
+# based on the enterprise-storage profile
+
+# ktune sysctl settings for rhel6 servers with enterprise-class
+# storage, maximizing i/o throughput
+#
+# Minimal preemption granularity for CPU-bound tasks:
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+kernel.sched_min_granularity_ns = 10000000
+
+# SCHED_OTHER wake-up granularity.
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+#
+# This option delays the preemption effects of decoupled workloads
+# and reduces their over-scheduling. Synchronous workloads will still
+# have immediate wakeup/sleep latencies.
+kernel.sched_wakeup_granularity_ns = 15000000
+
+# Filesystem I/O is usually much more efficient than swapping, so try to keep
+# swapping low.  It's usually safe to go even lower than this on systems with
+# server-grade storage.
+# vm.swappiness = 30
+
+# If a workload mostly uses anonymous memory and it hits this limit, the entire
+# working set is buffered for I/O, and any more write buffering would require
+# swapping, so it's time to throttle writes until I/O can catch up.  Workloads
+# that mostly use file mappings may be able to use even higher values.
+#
+# The generator of dirty data starts writeback at this percentage (system default
+# is 20%)
+vm.dirty_ratio = 40
+
+# Start background writeback (via writeback threads) at this percentage (system
+# default is 10%)
+#vm.dirty_background_ratio = 15
+
+# PID allocation wrap value.  When the kernel's next PID value
+# reaches this value, it wraps back to a minimum PID value.
+# PIDs of value pid_max or larger are not allocated.
+#
+# A suggested value for pid_max is 1024 * <# of cpu cores/threads in system>
+# e.g., a box with 32 cpus, the default of 32768 is reasonable, for 64 cpus,
+# 65536, for 4096 cpus, 4194304 (which is the upper limit possible).
+#kernel.pid_max = 65536
+
+# This is for HANA deployment
+kernel.sem = 1250 256000 100 8192
diff --git a/tune-profiles/sap-hana/tuned.conf b/tune-profiles/sap-hana/tuned.conf
new file mode 100644
index 0000000..d915459
--- /dev/null
+++ b/tune-profiles/sap-hana/tuned.conf
@@ -0,0 +1,22 @@
+# RHEL for SAP HANA tuned profile
+# based on the enterprise-storage profile
+
+[main]
+
+[DiskMonitor]
+enabled=False
+
+[DiskTuning]
+enabled=False
+
+[NetMonitor]
+enabled=False
+
+[NetTuning]
+enabled=False
+
+[CPUMonitor]
+enabled=False
+
+[CPUTuning]
+enabled=False
diff --git a/tune-profiles/sap-netweaver/ktune.sh b/tune-profiles/sap-netweaver/ktune.sh
new file mode 100644
index 0000000..8e84bfe
--- /dev/null
+++ b/tune-profiles/sap-netweaver/ktune.sh
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+# SAP ktune script
+# based on the enterprise-storage profile
+
+. /etc/tune-profiles/functions
+
+start() {
+	set_cpu_governor performance
+	set_transparent_hugepages always
+	disable_disk_barriers
+	multiply_disk_readahead 4
+
+	return 0
+}
+
+stop() {
+	restore_cpu_governor
+	restore_transparent_hugepages
+	enable_disk_barriers
+	restore_disk_readahead
+
+	return 0
+}
+
+process $@
diff --git a/tune-profiles/sap-netweaver/ktune.sysconfig b/tune-profiles/sap-netweaver/ktune.sysconfig
new file mode 100644
index 0000000..0950bfb
--- /dev/null
+++ b/tune-profiles/sap-netweaver/ktune.sysconfig
@@ -0,0 +1,42 @@
+# SAP ktune service configuration
+# based on the enterprise storage profile
+
+# This is the ktune sysctl file.  You can comment this out to prevent ktune
+# from applying its sysctl settings.
+SYSCTL="/etc/sysctl.ktune"
+
+# Use *.conf files in the ktune configuration directory /etc/ktune.d.
+#   Value: yes|no,  default: yes
+# It is useful if you want to load settings from additional files. Set this to
+# no if you to prevent ktune from using these additional files.
+USE_KTUNE_D="yes"
+
+# This is the custom sysctl configuration file. Any settings in this file will
+# be applied before the ktune settings, thus may be overridden by ktune.
+SYSCTL_PRE="/etc/sysctl.d/* /etc/sysctl.conf"
+
+# This is the custom sysctl configuration file. Any settings in this file will
+# be applied after the ktune settings, overriding them
+#SYSCTL_POST="/etc/sysctl.d/* /etc/sysctl.conf"
+
+# This is the I/O scheduler ktune will use. This will *not* override anything
+# explicitly set on the kernel command line, nor will it change the scheduler
+# for any block device that is using a non-default scheduler when ktune starts.
+# You should probably leave this on "deadline", but "as", "cfq", and "noop" are
+# also legal values.  Comment this out to prevent ktune from changing I/O
+# scheduler settings.
+ELEVATOR="deadline"
+
+# These are the devices, that should be tuned with the ELEVATOR
+ELEVATOR_TUNE_DEVS="/sys/block/{sd,cciss,dm-,vd}*/queue/scheduler"
+
+# The following lines are for autodetection of SAP settings
+SAP_MAIN_MEMORY_TOTAL=`awk '/MemTotal:/ {print $2}' /proc/meminfo`
+SAP_SWAP_SPACE_TOTAL=`awk '/SwapTotal:/ {print $2}' /proc/meminfo`
+# Rounding to full Gigabytes
+SAP_VIRT_MEMORY_TOTAL=$(( ( $SAP_MAIN_MEMORY_TOTAL + $SAP_SWAP_SPACE_TOTAL + 1048576 ) / 1048576 ))
+
+# kernel.shmall is in 4 KB pages; minimum 20 GB (SAP Note 941735)
+SAP_SHMALL=$(( $SAP_VIRT_MEMORY_TOTAL * 1024 * 1024 / 4 ))
+# kernel.shmmax is in Bytes; minimum 20 GB (SAP Note 941735)
+SAP_SHMMAX=$(( $SAP_VIRT_MEMORY_TOTAL * 1024 * 1024 * 1024 ))
diff --git a/tune-profiles/sap-netweaver/sysctl.ktune b/tune-profiles/sap-netweaver/sysctl.ktune
new file mode 100644
index 0000000..d1a4ec9
--- /dev/null
+++ b/tune-profiles/sap-netweaver/sysctl.ktune
@@ -0,0 +1,32 @@
+# SAP ktune sysctl settings
+# based on the enterprise-storage profile
+
+# ktune sysctl settings for rhel6 servers with enterprise-class
+# storage, maximizing i/o throughput
+#
+# Minimal preemption granularity for CPU-bound tasks:
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+kernel.sched_min_granularity_ns = 10000000
+
+# SCHED_OTHER wake-up granularity.
+# (default: 1 msec#  (1 + ilog(ncpus)), units: nanoseconds)
+#
+# This option delays the preemption effects of decoupled workloads
+# and reduces their over-scheduling. Synchronous workloads will still
+# have immediate wakeup/sleep latencies.
+kernel.sched_wakeup_granularity_ns = 15000000
+
+# If a workload mostly uses anonymous memory and it hits this limit, the entire
+# working set is buffered for I/O, and any more write buffering would require
+# swapping, so it's time to throttle writes until I/O can catch up.  Workloads
+# that mostly use file mappings may be able to use even higher values.
+#
+# The generator of dirty data starts writeback at this percentage (system default
+# is 20%)
+vm.dirty_ratio = 40
+
+# SAP specific tunings, SAP variables are preset in sysconfig
+kernel.shmall = >$SAP_SHMALL
+kernel.shmmax = >$SAP_SHMMAX
+kernel.sem = 1250 256000 100 8192
+vm.max_map_count = 1000000
diff --git a/tune-profiles/sap-netweaver/tuned.conf b/tune-profiles/sap-netweaver/tuned.conf
new file mode 100644
index 0000000..b4b3bfb
--- /dev/null
+++ b/tune-profiles/sap-netweaver/tuned.conf
@@ -0,0 +1,22 @@
+# SAP tuned profile
+# based on the enterprise-storage profile
+
+[main]
+
+[DiskMonitor]
+enabled=False
+
+[DiskTuning]
+enabled=False
+
+[NetMonitor]
+enabled=False
+
+[NetTuning]
+enabled=False
+
+[CPUMonitor]
+enabled=False
+
+[CPUTuning]
+enabled=False
-- 
1.9.3

