diff -up tuned-0.2.19/ktune/ktune.init.orig tuned-0.2.19/ktune/ktune.init
--- tuned-0.2.19/ktune/ktune.init.orig	2013-07-23 18:18:06.000000000 +0200
+++ tuned-0.2.19/ktune/ktune.init	2013-07-23 18:24:58.584351819 +0200
@@ -180,8 +180,6 @@ start() {
     [ "$EUID" != "0" ] && exit 4
     ret=0
 
-    load_sysctl
-
     # if no elevator on command line, apply the ktune elevator
     if [ -n "$ELEVATOR" -a -n "$SYS_BLOCK_SDX" -a $CMDLINE_ELEVATOR -ne 0 ]; then
 	ret_e=0
@@ -197,7 +195,9 @@ start() {
 	    failure; echo; ret=1
 	fi
     fi
-    
+
+    load_sysctl
+
     /bin/touch $VAR_SUBSYS_KTUNE
     return $ret
 }
