diff --git a/tuned-gui.glade b/tuned-gui.glade
index 36e173a..30dd51a 100644
--- a/tuned-gui.glade
+++ b/tuned-gui.glade
@@ -1,7 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Generated with glade 3.16.1 -->
 <interface>
-  <requires lib="gtk+" version="3.10"/>
   <object class="GtkAboutDialog" id="aboutdialog1">
     <property name="can_focus">False</property>
     <property name="type_hint">dialog</property>
@@ -729,12 +728,6 @@
                       <object class="GtkViewport" id="viewport1">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <child>
-                          <object class="GtkListBox" id="listboxSummaryOfActiveProfile">
-                            <property name="visible">True</property>
-                            <property name="can_focus">False</property>
-                          </object>
-                        </child>
                       </object>
                       <packing>
                         <property name="expand">False</property>
diff --git a/tuned-gui.py b/tuned-gui.py
index 8f72fd5..54fd815 100755
--- a/tuned-gui.py
+++ b/tuned-gui.py
@@ -320,8 +320,6 @@ class Base(object):
 		self.label_actual_profile.set_text(self.controller.active_profile())
 		if self.config.get(consts.CFG_RECOMMEND_COMMAND):
 			self.label_recommended_profile.set_text(self.controller.recommend_profile())
-		self.listbox_summary_of_active_profile = \
-			self.builder.get_object('listboxSummaryOfActiveProfile')
 
 		self.data_for_listbox_summary_of_active_profile()
 		self.comboboxtext_fast_change_profile.set_model(self.treestore_profiles)
@@ -446,9 +444,6 @@ class Base(object):
 		This method is emited after change profile and on startup of app.
 		"""
 
-		for row in self.listbox_summary_of_active_profile:
-			self.listbox_summary_of_active_profile.remove(row)
-
 		if self.is_tuned_connection_ok():
 			self.active_profile = \
 				self.manager.get_profile(self.controller.active_profile())
@@ -464,51 +459,6 @@ class Base(object):
 
 			self.label_summary_included_profile.set_text('None')
 
-		row = Gtk.ListBoxRow()
-		box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=0)
-		plugin_name = Gtk.Label()
-		plugin_name.set_markup('<b>Plugin Name</b>')
-		plugin_option = Gtk.Label()
-		plugin_option.set_markup('<b>Plugin Options</b>')
-		box.pack_start(plugin_name, True, True, 0)
-		box.pack_start(plugin_option, True, True, 0)
-		row.add(box)
-
-		self.listbox_summary_of_active_profile.add(row)
-
-		sep = Gtk.Separator.new(Gtk.Orientation.HORIZONTAL)
-		self.listbox_summary_of_active_profile.add(sep)
-		sep.show()
-
-		for u in self.active_profile.units:
-			row = Gtk.ListBoxRow()
-			hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL,
-						   spacing=0)
-			hbox.set_homogeneous(True)
-			row.add(hbox)
-			label = Gtk.Label()
-			label.set_markup(u)
-			label.set_justify(Gtk.Justification.LEFT)
-			hbox.pack_start(label, False, True, 1)
-
-			grid = Gtk.Box(orientation=Gtk.Orientation.VERTICAL,
-						   spacing=0)
-			grid.set_homogeneous(True)
-			for o in self.active_profile.units[u].options:
-				label_option = Gtk.Label()
-				label_option.set_markup(o + ' = ' + '<b>'
-						+ self.active_profile.units[u].options[o]
-						+ '</b>')
-				grid.pack_start(label_option, False, True, 0)
-
-			hbox.pack_start(grid, False, True, 0)
-			self.listbox_summary_of_active_profile.add(row)
-			separator = Gtk.Separator.new(Gtk.Orientation.HORIZONTAL)
-			self.listbox_summary_of_active_profile.add(separator)
-			separator.show()
-
-		self.listbox_summary_of_active_profile.show_all()
-
 	# def on_treeview_button_press_event(self, treeview, event):
 	#	 popup = Gtk.Menu()
 	#	 popup.append(Gtk.MenuItem('add'))
