diff --git a/profiles/cpu-partitioning/tuned.conf b/profiles/cpu-partitioning/tuned.conf
index 3c52215..1821b74 100644
--- a/profiles/cpu-partitioning/tuned.conf
+++ b/profiles/cpu-partitioning/tuned.conf
@@ -19,13 +19,13 @@ tmpdir=${f:strip:${f:exec:/usr/bin/mktemp:-d}}
 isolated_cores_expanded=${f:cpulist_unpack:${isolated_cores}}
 isolated_cpumask=${f:cpulist2hex:${isolated_cores_expanded}}
 not_isolated_cores_expanded=${f:cpulist_invert:${isolated_cores_expanded}}
-isolated_cores_present_expanded=${f:cpulist_present:${isolated_cores}}
-not_isolated_cores_present_expanded=${f:cpulist_present:${not_isolated_cores_expanded}}
+isolated_cores_online_expanded=${f:cpulist_online:${isolated_cores}}
+not_isolated_cores_online_expanded=${f:cpulist_online:${not_isolated_cores_expanded}}
 not_isolated_cpumask=${f:cpulist2hex:${not_isolated_cores_expanded}}
 no_balance_cores_expanded=${f:cpulist_unpack:${no_balance_cores}}
 
-# Fail if isolated_cores contains CPUs which are not present
-assert2=${f:assertion:isolated_cores contains present CPU(s):${isolated_cores_expanded}:${isolated_cores_present_expanded}}
+# Fail if isolated_cores contains CPUs which are not online
+assert2=${f:assertion:isolated_cores contains online CPU(s):${isolated_cores_expanded}:${isolated_cores_online_expanded}}
 
 [sysctl]
 kernel.hung_task_timeout_secs = 600
diff --git a/profiles/realtime-virtual-guest/tuned.conf b/profiles/realtime-virtual-guest/tuned.conf
index b90e76f..fb2bc42 100644
--- a/profiles/realtime-virtual-guest/tuned.conf
+++ b/profiles/realtime-virtual-guest/tuned.conf
@@ -15,10 +15,10 @@ isolated_cores_assert_check = \\${isolated_cores}
 assert1=${f:assertion_non_equal:isolated_cores are set:${isolated_cores}:${isolated_cores_assert_check}}
 
 isolated_cores_expanded=${f:cpulist_unpack:${isolated_cores}}
-isolated_cores_present_expanded=${f:cpulist_present:${isolated_cores}}
+isolated_cores_online_expanded=${f:cpulist_online:${isolated_cores}}
 
-# Fail if isolated_cores contains CPUs which are not present
-assert2=${f:assertion:isolated_cores contains present CPU(s):${isolated_cores_expanded}:${isolated_cores_present_expanded}}
+# Fail if isolated_cores contains CPUs which are not online
+assert2=${f:assertion:isolated_cores contains online CPU(s):${isolated_cores_expanded}:${isolated_cores_online_expanded}}
 
 [scheduler]
 # group.group_name=rule_priority:scheduler_policy:scheduler_priority:core_affinity_in_hex:process_name_regex
diff --git a/profiles/realtime-virtual-host/tuned.conf b/profiles/realtime-virtual-host/tuned.conf
index 0346fff..5e0ff1f 100644
--- a/profiles/realtime-virtual-host/tuned.conf
+++ b/profiles/realtime-virtual-host/tuned.conf
@@ -20,10 +20,10 @@ isolated_cores_assert_check = \\${isolated_cores}
 assert1=${f:assertion_non_equal:isolated_cores are set:${isolated_cores}:${isolated_cores_assert_check}}
 
 isolated_cores_expanded=${f:cpulist_unpack:${isolated_cores}}
-isolated_cores_present_expanded=${f:cpulist_present:${isolated_cores}}
+isolated_cores_online_expanded=${f:cpulist_online:${isolated_cores}}
 
-# Fail if isolated_cores contains CPUs which are not present
-assert2=${f:assertion:isolated_cores contains present CPU(s):${isolated_cores_expanded}:${isolated_cores_present_expanded}}
+# Fail if isolated_cores contains CPUs which are not online
+assert2=${f:assertion:isolated_cores contains online CPU(s):${isolated_cores_expanded}:${isolated_cores_online_expanded}}
 
 [scheduler]
 # group.group_name=rule_priority:scheduler_policy:scheduler_priority:core_affinity_in_hex:process_name_regex
diff --git a/profiles/realtime/tuned.conf b/profiles/realtime/tuned.conf
index b2273cf..c595e67 100644
--- a/profiles/realtime/tuned.conf
+++ b/profiles/realtime/tuned.conf
@@ -18,10 +18,10 @@ assert1=${f:assertion_non_equal:isolated_cores are set:${isolated_cores}:${isola
 # Non-isolated cores cpumask including offline cores
 not_isolated_cpumask = ${f:cpulist2hex_invert:${isolated_cores}}
 isolated_cores_expanded=${f:cpulist_unpack:${isolated_cores}}
-isolated_cores_present_expanded=${f:cpulist_present:${isolated_cores}}
+isolated_cores_online_expanded=${f:cpulist_online:${isolated_cores}}
 
-# Fail if isolated_cores contains CPUs which are not present
-assert2=${f:assertion:isolated_cores contains present CPU(s):${isolated_cores_expanded}:${isolated_cores_present_expanded}}
+# Fail if isolated_cores contains CPUs which are not online
+assert2=${f:assertion:isolated_cores contains online CPU(s):${isolated_cores_expanded}:${isolated_cores_online_expanded}}
 
 [sysctl]
 kernel.hung_task_timeout_secs = 600
diff --git a/tuned/profiles/functions/function_cpulist_invert.py b/tuned/profiles/functions/function_cpulist_invert.py
index 375eb67..2aff3c9 100644
--- a/tuned/profiles/functions/function_cpulist_invert.py
+++ b/tuned/profiles/functions/function_cpulist_invert.py
@@ -8,7 +8,7 @@ log = tuned.logs.get()
 class cpulist_invert(base.Function):
 	"""
 	Inverts list of CPUs (makes its complement). For the complement it
-	gets number of present CPUs from the /sys/devices/system/cpu/present,
+	gets number of present CPUs from the /sys/devices/system/cpu/online,
 	e.g. system with 4 CPUs (0-3), the inversion of list "0,2,3" will be
 	"1"
 	"""
diff --git a/tuned/utils/commands.py b/tuned/utils/commands.py
index 8b7df57..41d6d99 100644
--- a/tuned/utils/commands.py
+++ b/tuned/utils/commands.py
@@ -363,8 +363,8 @@ class commands:
 	# Inverts CPU list (i.e. makes its complement)
 	def cpulist_invert(self, l):
 		cpus = self.cpulist_unpack(l)
-		present = self.cpulist_unpack(self.read_file("/sys/devices/system/cpu/present"))
-		return list(set(present) - set(cpus))
+		online = self.cpulist_unpack(self.read_file("/sys/devices/system/cpu/online"))
+		return list(set(online) - set(cpus))
 
 	# Converts CPU list to hexadecimal CPU mask
 	def cpulist2hex(self, l):
