diff -up tuned-0.2.19/tune-profiles/functions.orig tuned-0.2.19/tune-profiles/functions
--- tuned-0.2.19/tune-profiles/functions.orig	2013-07-23 16:38:15.000000000 +0200
+++ tuned-0.2.19/tune-profiles/functions	2013-07-23 17:33:29.901918000 +0200
@@ -279,13 +279,13 @@ disable_wifi_powersave() {
 #
 
 disable_bluetooth() {
-	hciconfig hci0 down
-	rmmod hci_usb
+	hciconfig hci0 down >/dev/null 2>&1
+	lsmod | grep -q btusb && rmmod btusb
 }
 
 enable_bluetooth() {
-	modprobe hci_usb
-	hciconfig hci0 up
+	modprobe btusb
+	hciconfig hci0 up >/dev/null 2>&1
 }
 
 #
@@ -297,7 +297,7 @@ _usb_autosuspend() {
 	for i in /sys/bus/usb/devices/*/power/autosuspend; do echo $value > $i; done &> /dev/null
 }
 
-enable_usb_autosupend() {
+enable_usb_autosuspend() {
 	_usb_autosuspend 1
 }
 
